packages/amplify-graphql-auth-transformer/src/utils/index.ts

```diff
@@ -0,0 +1,101 @@
+import { toCamelCase } from 'graphql-transformer-common';
+import { AuthProvider, AuthRule } from './definitions';
+
+export * from './constants';
+export * from './definitions';
+export * from './validations';
+
+/**
+ * Ensure the following defaults
+ * - provider
+ * - iam policy generation
+ */
+export const ensureAuthRuleDefaults = (rules: AuthRule[]) => {
+  // We assign the default provider if an override is not present make further handling easier.
+  for (const rule of rules) {
+    if (!rule.provider) {
+      switch (rule.allow) {
+        case 'owner':
+        case 'groups':
+          rule.provider = 'userPools';
+          break;
+        case 'private':
+          rule.provider = 'userPools';
+          break;
+        case 'public':
+          rule.provider = 'apiKey';
+          break;
+        default:
+          rule.provider = null;
+          break;
+      }
+    }
+    if (rule.provider === 'iam') {
+      rule.generateIAMPolicy = true;
+    }
+  }
+};
+
+export const getModelConfig = (directive: DirectiveNode, typeName: string): ModelDirectiveConfiguration => {
+  const directiveWrapped: DirectiveWrapper = new DirectiveWrapper(directive);
+  const options = directiveWrapped.getArguments<ModelDirectiveConfiguration>({
+    queries: {
+      get: toCamelCase(['get', typeName]),
+      list: toCamelCase(['list', `${typeName}s`]),
+    },
+    mutations: {
+      create: toCamelCase(['create', typeName]),
+      update: toCamelCase(['update', typeName]),
+      delete: toCamelCase(['delete', typeName]),
+    },
+    subscriptions: {
+      level: SubscriptionLevel.public,
+      onCreate: toCamelCase(['onCreate', typeName]),
+      onDelete: toCamelCase(['onDelete', typeName]),
+      onUpdate: toCamelCase(['onUpdate', typeName]),
+    },
+    timestamps: {
+      createdAt: 'createdAt',
+      updatedAt: 'updatedAt',
+    },
+  });
+  return options;
+};
+
+export const getConfiguredAuthProviders = (authConfig: AppSyncAuthConfiguration) => {
+  const providers = [
+    authConfig.defaultAuthentication.authenticationType,
+    ...authConfig.additionalAuthenticationProviders.map(p => p.authenticationType),
+  ];
+  const getAuthProvider = (authType: AppSyncAuthMode): AuthProvider => {
+    switch (authType) {
+      case 'AMAZON_COGNITO_USER_POOLS':
+        return 'userPools';
+      case 'API_KEY':
+        return 'apiKey';
+      case 'AWS_IAM':
+        return 'iam';
+      case 'OPENID_CONNECT':
+        return 'oidc';
+    }
+  };
+
+  return {
+    default: getAuthProvider(authConfig.defaultAuthentication.authenticationType),
+    onlyDefaultAuthProviderConfigured: authConfig.additionalAuthenticationProviders.length === 0,
+    hasApiKey: providers.find(p => p === 'API_KEY') ? true : false,
+    hasUserPools: providers.find(p => p === 'AMAZON_COGNITO_USER_POOLS') ? true : false,
+    hasOIDC: providers.find(p => p === 'OPENID_CONNECT') ? true : false,
+    hasIAM: providers.find(p => p === 'AWS_IAM') ? true : false,
```

Comment:
Since the return value from `find()` isn't used, these can be simplified to just, for example:

```js
hasApiKey: providers.some(p => p === 'API_KEY'),
```